lib/iris/_merge.py

```diff
@@ -1228,17 +1227,23 @@ def merge(self, unique=True):
                 data = self._skeletons[group[offset]].data
                 # Ensure the data is represented as a dask array and
                 # slot that array into the stack.
-                if is_lazy_data(data):
+                if iris.util.is_dask_array(data):
                     all_have_data = False
                 else:
-                    data = as_lazy_data(data)
+                    if isinstance(data, np.ma.MaskedArray):
+                        if np.ma.is_masked(data):
+                            if data.dtype.kind == 'i':
+                                data = data.astype('f8')
+                            data[data.mask] = np.nan
+                        data = data.data
```

Comment:
@pp-mo and i had a similar conversation about this pattern.  I think there is a neater abstraction to be had.

please may we continue with this slightly awkward implementation whilst this PR gets test hardy, then implement an improved pattern as a follow up?
perhaps you could raise a ticket for this for us?